src/libraries/System.Private.Uri/src/System/UriExt.cs

```diff
@@ -338,13 +338,14 @@ public string GetComponents(UriComponents components, UriFormat format)
         public static int Compare(Uri? uri1, Uri? uri2, UriComponents partsToCompare, UriFormat compareFormat,
             StringComparison comparisonType)
         {
-            if ((object?)uri1 == null)
+            if (uri1 is null)
             {
-                if (uri2 == null)
+                if (uri2 is null)
```

Comment:
Here there was no cast to object, but you still made the change. What is the rule?

Reply:
My rule was "apply the fix to each line individually, validating it was doing the right thing, and then also fix for consistency anything immediately above or below I happened to see as I was doing so"